pkg/cmd/infra/deployer/deployer.go

```diff
@@ -38,7 +47,20 @@ func NewCommandDeployer(name string) *cobra.Command {
 		Short: "Run the OpenShift deployer",
 		Long:  longCommandDesc,
 		Run: func(c *cobra.Command, args []string) {
-			if err := deploy(cfg); err != nil {
+			kClient, _, err := cfg.Config.Clients()
+			if err != nil {
+				glog.Fatal(err)
+			}
+
+			if len(cfg.DeploymentName) == 0 {
+				glog.Fatal("deployment is required")
+			}
+
+			if len(cfg.Namespace) == 0 {
```

Comment:
The new deployer logic goes in lockstep with rollbacks- it now disables rather than deleting RCs, and has all the extra stuff to deal with the fact that they are still present and must be filtered.